// fetch-wrapper/lib/index.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.sendRequest = exports.getAuthReq = exports.putAuthReq = exports.postAuthReq = exports.putReq = exports.getReq = exports.postReq = undefined;

var _requestCreatorHelpers = require('./request-creator-helpers.js');

var requestHelpers = _interopRequireWildcard(_requestCreatorHelpers);

var _fetchWrapper = require('./fetch-wrapper.js');

var _fetchWrapper2 = _interopRequireDefault(_fetchWrapper);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } else { var newObj = {}; if (obj != null) { for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) newObj[key] = obj[key]; } } newObj.default = obj; return newObj; } }

var postReq = exports.postReq = requestHelpers.postReq;
var getReq = exports.getReq = requestHelpers.getReq;
var putReq = exports.putReq = requestHelpers.putReq;
var postAuthReq = exports.postAuthReq = requestHelpers.postAuthReq;
var putAuthReq = exports.putAuthReq = requestHelpers.putAuthReq;
var getAuthReq = exports.getAuthReq = requestHelpers.getAuthReq;
var sendRequest = exports.sendRequest = _fetchWrapper2.default;

/**
EXAMPLE USAGE

sendRequest({
  request: postReq('localhost:9009/login', {name: 'name'}), //this should be a function that returns a fetch request
  responseType: 'json'
  onSuccess: json => {},
  onError: (error) => {}
})
**/
